core/data_array.cpp

```diff
@@ -113,26 +118,55 @@ Variable sparse_dense_op_impl(Op op, const VariableConstProxy &sparseCoord_,
 
 DataArray &DataArray::operator+=(const DataConstProxy &other) {
   expect::coordsAndLabelsAreSuperset(*this, other);
+  union_or_in_place(masks(), other.masks());
   data() += other.data();
   return *this;
 }
 
 DataArray &DataArray::operator-=(const DataConstProxy &other) {
   expect::coordsAndLabelsAreSuperset(*this, other);
+  union_or_in_place(masks(), other.masks());
   data() -= other.data();
   return *this;
 }
 
+template <class Op>
+DataArray &sparse_dense_op_inplace(Op op, DataArray &a,
+                                   const DataConstProxy &b) {
+  if (!is_sparse_and_histogram(a, b)) {
+    expect::coordsAndLabelsAreSuperset(a, b);
+    union_or_in_place(a.masks(), b.masks());
+    std::is_same_v<Op, std::decay_t<decltype(times)>> ? a.data() *= b.data()
```

Comment:
This seems to imply that template arg `Op` is either times or divide. i.e could be misused for other operation arguments - would it be better to redefine `times` and `divide` to be structs that could provide the operators required by this function?